### PR TITLE
DXCDT-315: Refactor log streams create and update cmds

### PR DESCRIPTION
**Pull request by sergiught**
### 🔧 Changes

Refactoring `auth0 logs streams create|update` to have an extra subcommand to create a specific log stream type that only has a subset of the flags actually needed for that log stream e.g. `auth0 logs streams create datadog [flags]`.

https://user-images.githubusercontent.com/28300158/211884934-05cc3543-5c45-42b6-92b0-556b922d06f0.mov

### 📝 Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

**Comment by Widcket** [comment]
Shouldn't the `--json` flag result in colored output, unless `--no-color` is specified?

Screenshot 2023-01-11 at 15 18 26

**Comment by Widcket** [comment]
I notice in some commands this `%` symbol, seems like a renderer method is missing a `\n`.

Screenshot 2023-01-11 at 15 20 09

**Comment by sergiught** [comment]
> Shouldn't the --json flag result in colored output, unless --no-color is specified?

@Widcket: Good catch! That should be indeed the case. I wonder if https://github.com/auth0/auth0-cli/pull/594 had unintended consequences. We'll fix this in a separate PR as it's unrelated to the log streams command restructuring. Thanks for spotting this! 

> I notice in some commands this % symbol, seems like a renderer method is missing a \n.

IMO when `--json` flag is present we shouldn't end with a new line as that will mess up the input for tools like `jq`. 

This is how it will look like with an additional new line after the json output (to remove the % symbol):

image

This is how it looks like without the new line: 

image

**Comment by Widcket** [comment]
> IMO when --json flag is present we shouldn't end with a new line as that will mess up the input for tools like jq.

Yes, that's right.

File: internal/cli/log_streams.go [hunk 1]
```diff
@@ -241,486 +130,38 @@ func showLogStreamCmd(cli *cli) *cobra.Command {
...
-			}); err != nil {
-				return fmt.Errorf("Unable to update log stream: %v", err)
-			}
-
-			// Render log stream update specific view
-			cli.renderer.LogStreamUpdate(ls)
-			return nil
-		},
+		Short: "Update an existing log stream",
+		Long:  "",
```
Comment (Widcket) [inline internal/cli/log_streams.go:155]: This long description is missing.

**Review by Widcket — APPROVED** [review]

File: internal/cli/log_streams.go [hunk 2]
```diff
@@ -159,7 +48,7 @@ func logStreamsCmd(cli *cli) *cobra.Command {
 	cmd.SetUsageTemplate(resourceUsageTemplate())
 	cmd.AddCommand(listLogStreamsCmd(cli))
 	cmd.AddCommand(showLogStreamCmd(cli))
-	cmd.AddCommand(createLogStreamCmd(cli))
+	cmd.AddCommand(createLogStreamsCmd(cli))
```
Comment (willvedd) [inline internal/cli/log_streams.go:51]:
```suggestion
	cmd.AddCommand(createLogStreamCmd(cli))
```

File: internal/cli/log_streams_datadog.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  auth0 logs streams create datadog --name <name> --region <region>
+  auth0 logs streams create datadog --name <name> --region <region> --api-key <api-key>
+  auth0 logs streams create datadog -n <name> -r <region> -k <api-key>
+  auth0 logs streams create datadog -n mylogstream -r eu -k 121233123455 --json`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := logStreamName.Ask(cmd, &inputs.Name, nil); err != nil {
+				return err
+			}
+
+			if err := datadogRegion.Ask(cmd, &inputs.DatadogRegion, nil); err != nil {
```
Comment (willvedd) [inline internal/cli/log_streams_datadog.go:56]: Maybe provide a survey of available options?

File: internal/cli/log_streams_http.go [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+		LongForm:     "type",
+		ShortForm:    "t",
+		Help:         "The \"Content-Type\" header to send over HTTP. Common value is \"application/json\".",
+		AlwaysPrompt: true,
+	}
+	httpContentFormat = Flag{
+		Name:         "HTTP Content Format",
+		LongForm:     "format",
+		ShortForm:    "f",
+		Help:         "The format of data sent over HTTP. Options are \"JSONLINES\", \"JSONARRAY\" or \"JSONOBJECT\"",
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/log_streams_http.go:33]: Fixed in [`b509205` (#599)](https://github.com/auth0/auth0-cli/pull/599/commits/b509205955d9015aa54a79350ec6f17c1db76b96)

File: internal/cli/log_streams_event_bridge.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+	"github.com/auth0/auth0-cli/internal/ansi"
+	"github.com/auth0/auth0-cli/internal/auth0"
+)
+
+var (
+	awsAccountID = Flag{
+		Name:       "AWS Account ID",
+		LongForm:   "aws-id",
+		ShortForm:  "i",
+		Help:       "Id of the AWS account.",
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/log_streams_event_bridge.go]: Fixed in [`b509205` (#599)](https://github.com/auth0/auth0-cli/pull/599/commits/b509205955d9015aa54a79350ec6f17c1db76b96)

File: internal/cli/log_streams_datadog.go (hunk 3 above)
Comment (sergiught) [reply internal/cli/log_streams_datadog.go]: Fixed in [`b509205` (#599)](https://github.com/auth0/auth0-cli/pull/599/commits/b509205955d9015aa54a79350ec6f17c1db76b96)

File: internal/cli/log_streams.go (hunk 2 above)
Comment (sergiught) [reply internal/cli/log_streams.go]: Fixed in [`b509205` (#599)](https://github.com/auth0/auth0-cli/pull/599/commits/b509205955d9015aa54a79350ec6f17c1db76b96)

File: internal/cli/log_streams.go [hunk 6]
```diff
@@ -241,486 +130,38 @@ func showLogStreamCmd(cli *cli) *cobra.Command {
...
-				return cli.api.LogStream.Create(ls)
-			}); err != nil {
-				return fmt.Errorf("Unable to create log stream: %v", err)
-			}
-
-			// Render log stream creation specific view
-			cli.renderer.LogStreamCreate(ls)
-			return nil
-		},
+		Long:  "Log Streaming allows you to export your events in near real-time. Choose an option below to configure.",
```
[the remark this answers is not in the document]
Comment (sergiught) [reply internal/cli/log_streams.go]: Fixed in [`b509205` (#599)](https://github.com/auth0/auth0-cli/pull/599/commits/b509205955d9015aa54a79350ec6f17c1db76b96)

**Pull request merged**